bittensor/subtensor.py

```diff
@@ -4456,10 +4470,17 @@ async def get_delegates(self, block: Optional[int] = None) -> List[DelegateInfo]
 
         """
 
-        @retry(stop=stop_after_attempt(3), wait=wait_fixed(1))
+        @retry(
+            stop=stop_after_attempt(3),
+            wait=wait_fixed(1),
+            retry=retry_if_exception_type(SubstrateRequestException),
```

Comment:
You're actually going to want to use `retry=retry_if_not_exception_type(SubstrateRequestException)` rather than `retry_if_exception_type`, because otherwise it will never raise `SubstrateRequestException`.

Reply:
Sounds reasonable. I applied this in a fix to this project, which I am now preparing for review.